app/src/main/java/com/github/se/studybuddies/database/DatabaseConnection.kt

```diff
@@ -312,4 +321,84 @@ class DatabaseConnection {
   fun getGroupMessagesPath(groupUID: String): String {
     return MessageVal.GROUPS + "/$groupUID/" + MessageVal.MESSAGES
   }
+
+  fun updateTodo(
+      todoId: String,
+      name: String,
+      assigneeName: String,
+      dueDate: Date,
+      location: String,
+      description: String,
+      status: String
+  ) {
+    val task =
+        hashMapOf(
+            "title" to name,
+            "assigneeName" to assigneeName,
+            "dueDate" to dueDate,
+            "location" to location,
+            "description" to description,
+            "status" to status)
+    todoCollection
+        .document(todoId)
+        .update(task as Map<String, Any>)
+        .addOnSuccessListener { Log.d("MyPrint", "Task $todoId succesfully updated") }
+        .addOnFailureListener { Log.d("MyPrint", "Task $todoId failed to update") }
+  }
+
+  suspend fun getAllItems(): ToDoList {
+    val querySnapshot = todoCollection.get().await()
+    val items = mutableListOf<ToDo>()
+
+    for (document in querySnapshot.documents) {
+      val uid = document.id
+      val name = document.getString("title") ?: ""
+      val assigneeName = document.getString("assigneeName") ?: ""
+      val dueDate = document.getDate("dueDate")
+      val convertedDate = dueDate!!.toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
+      val locationString = document.getString("location") ?: ""
+      val location = Location.fromString(locationString)
+      val description = document.getString("description") ?: ""
+      val status = ToDoStatus.valueOf(document.getString("status") ?: "")
+
+      val item = ToDo(uid, name, assigneeName, convertedDate, location, description, status)
+      items.add(item)
+    }
+
+    return ToDoList(items)
+  }
+
+  fun addNewTodo(
+      name: String,
+      assigneeName: String,
+      dueDate: Date,
+      location: String,
+      description: String,
+      status: String
+  ) {
+    val task =
+        hashMapOf(
+            "title" to name,
+            "assigneeName" to assigneeName,
+            "dueDate" to dueDate,
+            "location" to location,
+            "description" to description,
+            "status" to status)
+    todoCollection
+        .add(task)
+        .addOnSuccessListener { Log.d("MyPrint", "Task succesfully added") }
+        .addOnFailureListener { Log.d("MyPrint", "Failed to add task") }
+  }
+
+  fun fetchTaskByUID(uid: String): Task<DocumentSnapshot> {
+    return todoCollection.document(uid).get()
+  }
+
+  fun deleteTodo(todoId: String) {
+    todoCollection
+        .document(todoId)
+        .delete()
+        .addOnSuccessListener { Log.d("MyPrint", "Successfully deleted task") }
+        .addOnFailureListener { Log.d("MyPrint", "Failed to delete task") }
+  }
```

Comment:
I'm not really proficient on FireBase but your code doesn't seem to work. I test you app on my tablet and tried to create a new Todo task and it didn't work ;-;. I would advise you to watch this video https://www.youtube.com/watch?v=bOd3wO0uFr8&t=1157s
Or I can also share you my bootcamp where I implemented the todo list (I will try to add it on a public git soon)